modules/aruco/misc/pattern_generator/MarkerPrinterGUI.py

```diff
@@ -0,0 +1,827 @@
+from MarkerPrinter import *
+
+import tkinter as tk
+from tkinter import ttk
+from tkinter import filedialog
+from tkinter import messagebox
+
+import time
+
+import PIL.Image
+import PIL.ImageTk
+
+class MarkerPrinterGUI:
+
+    def VisDPI(self, shape):
+        scale0 = float(self.displayShape[0]) / float(shape[0])
+        scale1 = float(self.displayShape[1]) / float(shape[1])
+
+        if(scale0 > scale1):
+            return scale1 * 96.0
+        else:
+            return scale0 * 96.0
+
+    def OnShowingHelpGithub(self):
+        messagebox.showinfo("Github",
+            "https://github.com/dogod621/OpenCVMarkerPrinter")
+
+    def OnCloseWindow(self):
+        if(self.window is not None):
+            if messagebox.askokcancel("Quit", "Do you want to quit?"):
+                self.window.destroy()
+                self.window = None
+
+    def OnSelectCharucoMarkerDictionary(self, pDictName):
+        self.charucoMarkerDictionaryStr.set(pDictName)
+
+    def OnPreviewOrSaveCharucoMarker(self, askSave = False):
+        sizeX = None
+        try:
+            sizeX = int(self.charucoMarkerChessboardSizeXStr.get())
```

Comment:
I believe this whole code should be written more compact:
```.py
try:
    do1
    do2
    do3
except TypeError as e:
    messagebox.showinfo("...")
except Exception as e:
    messagebox.showinfo("...")
```
Or maybe each `try..catch` should be extracted to a function accepting lambda expression and error message.

Reply:
Ok~ I feel this part of code is very very redundant too